### PR TITLE
Sidebar service

### PR DESCRIPTION
**Pull request by crystallow1168**
Hi Kevin,
Please review my PR, Thanks!

### CODE REVIEW

File: client/src/components/Hours.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    super(props);
+
+    this.state = {
+      hoursToggle: true,
+    };
+  }
+
+  handleHoursToggle(e) {
+    e.preventDefault();
+    const { hoursToggle } = this.state;
```
Comment (kkaiyang94) [inline client/src/components/Hours.jsx:15]: Don't need to destructure this here

File: client/src/components/Hours.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    this.setState({
+      hoursToggle: !hoursToggle,
+    });
+  }
+
+  render() {
+    const { hoursToggle } = this.state;
+    const { openHours, closeHours } = this.props;
```
Comment (kkaiyang94) [inline client/src/components/Hours.jsx:23]: You can destructure this portion as:

const { 
  openHours: { MonOH, TuesOH, WedOH, ThursOH, FriOH, SatOH, SunOH },
 closeHours: { MonCH, TuesCH, WedCH, ThursCH, FriCH, SatCH, SunCH },
         } = this.props

File: client/src/components/SideBar.jsx [hunk 3]
```diff
@@ -1,37 +1,49 @@
...
+    super(props);
+
+    this.state = {
+    };
+  }
+
+  render() {
+    const { details } = this.props;
+    const {
+      openHours, closeHours, address, phoneNumber, officialWebsite,
```
Comment (kkaiyang94) [inline client/src/components/SideBar.jsx:16]: Nothing wrong with the destructuring here, but we can do a little bit more.

**Review by kkaiyang94 — APPROVED** [review]
Overall the code looks very good! I think if we were able to add a mapping function to list all the hours the restaurant was open it would be nice, but the functionality as is works perfectly fine.

File: client/src/components/Hours.jsx (hunk 2 above)
Comment (crystallow1168) [reply client/src/components/Hours.jsx:23]: Oh! I didn't know I can do it like that too! Thanks!

File: client/src/components/Hours.jsx (hunk 1 above)
Comment (crystallow1168) [reply client/src/components/Hours.jsx:15]: The linter is making me de-structuring it!

File: client/src/components/SideBar.jsx (hunk 3 above)
Comment (crystallow1168) [reply client/src/components/SideBar.jsx:16]: I refactored it!

**Pull request merged**